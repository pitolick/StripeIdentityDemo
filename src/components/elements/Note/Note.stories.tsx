import React from 'react'
import { Meta, StoryObj } from '@storybook/react'
import { Note } from './Note'

const meta: Meta<typeof Note> = {
  title: 'elements/Note',
  component: Note,
  tags: ['autodocs'],
}

export default meta
type Story = StoryObj<typeof Note>

/**
 * ulリスト
 */
export const Ul: Story = {
  args: {
    list: ['注釈1', '注釈2'],
  },
}

/**
 * olリスト
 */
export const Ol: Story = {
  args: {
    tag: 'ol',
    list: ['注釈1', '注釈2'],
  },
}

/**
 * リストなし
 */
export const NoList: Story = {
  args: {
    list: [],
  },
}

/**
 * リンクありリスト
 */
export const LinkList: Story = {
  args: {
    list: [
      <>
        注釈注釈
        <a className="text-blue underline" href="https://www.google.com/" target="_blank" rel="noopener noreferrer">
          リンク1
        </a>
        注釈注釈
      </>,
      <>
        注釈注釈
        <br />
        <a className="text-blue underline" href="https://www.google.com/" target="_blank" rel="noopener noreferrer">
          リンク2
        </a>
        注釈注釈
      </>,
    ],
  },
}
