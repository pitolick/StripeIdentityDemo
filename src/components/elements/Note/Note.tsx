/**
 * 注釈コンポーネント
 * @module Note
 */
import React from 'react'

/**
 * 注釈コンポーネントのプロパティ
 * @typedef {Object} NoteProps
 */
export interface NoteProps {
  /**
   * タグ
   */
  tag?: 'ul' | 'ol'
  /**
   * ループ可能な子要素のリスト
   */
  list: React.ReactNode[]
}

/**
 * 注釈コンポーネント
 * @param {NoteProps} props - 注釈コンポーネントのプロパティ
 * @returns {React.FC} - 注釈コンポーネント
 */
export const Note: React.FC<NoteProps> = ({ tag = 'ul', list }) => {
  // listが空の場合は空の要素を返す
  if (!list || list.length === 0) {
    return <></>
  }

  const Tag = tag
  let counterClass = "ps-4 -indent-4 before:content-['※']"
  if (tag === 'ol') {
    counterClass = 'ps-4 -indent-4 note-ol'
  }
  const lists = list.map((item, index) => {
    return (
      <li className={`text-xs text-gray-700 before:me-1 ${counterClass}`} key={'note_' + index}>
        {item}
      </li>
    )
  })
  return <Tag>{lists}</Tag>
}
