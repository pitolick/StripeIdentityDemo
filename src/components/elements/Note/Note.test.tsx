import React from 'react'
import { render } from '@testing-library/react'
import { Note, NoteProps } from './Note'

describe('Noteコンポーネント', () => {
  const defaultProps: NoteProps = {
    list: ['注釈1', '注釈2', '注釈3'],
  }

  it('正しくレンダリングされること', () => {
    const { container } = render(<Note {...defaultProps} />)
    expect(container).toMatchSnapshot()
  })

  it('listが空の場合、空の要素が返されること', () => {
    const { container } = render(<Note list={[]} />)
    expect(container).toMatchSnapshot()
  })

  it('tagが"ol"の場合、正しいクラスが設定されること', () => {
    const { container } = render(<Note {...defaultProps} tag="ol" />)
    expect(container).toMatchSnapshot()
  })

  it('tagが"ul"の場合、正しいクラスが設定されること', () => {
    const { container } = render(<Note {...defaultProps} tag="ul" />)
    expect(container).toMatchSnapshot()
  })
})
