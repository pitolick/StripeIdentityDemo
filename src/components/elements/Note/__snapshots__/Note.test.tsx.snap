// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Noteコンポーネント listが空の場合、空の要素が返されること 1`] = `<div />`;

exports[`Noteコンポーネント tagが"ol"の場合、正しいクラスが設定されること 1`] = `
<div>
  <ol>
    <li
      class="text-xs text-gray-700 before:me-1 ps-4 -indent-4 note-ol"
    >
      注釈1
    </li>
    <li
      class="text-xs text-gray-700 before:me-1 ps-4 -indent-4 note-ol"
    >
      注釈2
    </li>
    <li
      class="text-xs text-gray-700 before:me-1 ps-4 -indent-4 note-ol"
    >
      注釈3
    </li>
  </ol>
</div>
`;

exports[`Noteコンポーネント tagが"ul"の場合、正しいクラスが設定されること 1`] = `
<div>
  <ul>
    <li
      class="text-xs text-gray-700 before:me-1 ps-4 -indent-4 before:content-['※']"
    >
      注釈1
    </li>
    <li
      class="text-xs text-gray-700 before:me-1 ps-4 -indent-4 before:content-['※']"
    >
      注釈2
    </li>
    <li
      class="text-xs text-gray-700 before:me-1 ps-4 -indent-4 before:content-['※']"
    >
      注釈3
    </li>
  </ul>
</div>
`;

exports[`Noteコンポーネント 正しくレンダリングされること 1`] = `
<div>
  <ul>
    <li
      class="text-xs text-gray-700 before:me-1 ps-4 -indent-4 before:content-['※']"
    >
      注釈1
    </li>
    <li
      class="text-xs text-gray-700 before:me-1 ps-4 -indent-4 before:content-['※']"
    >
      注釈2
    </li>
    <li
      class="text-xs text-gray-700 before:me-1 ps-4 -indent-4 before:content-['※']"
    >
      注釈3
    </li>
  </ul>
</div>
`;
